go/vt/vtadmin/http/workflows.go

```diff
@@ -0,0 +1,71 @@
+/*
+Copyright 2021 The Vitess Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+	http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package http
+
+import (
+	"context"
+
+	vtadminpb "vitess.io/vitess/go/vt/proto/vtadmin"
+)
+
+// GetWorkflow implements the http wrapper for the VTAdminServer.GetWorkflow
+// method.
+//
+// Its route is /workflow/{cluster_id}/{keyspace}/{name}[?active_only=].
+func GetWorkflow(ctx context.Context, r Request, api *API) *JSONResponse {
+	vars := r.Vars()
+
+	activeOnly, err := r.ParseQueryParamAsBool("active_only", false)
+	if err != nil {
+		return NewJSONResponse(nil, err)
+	}
+
+	workflow, err := api.server.GetWorkflow(ctx, &vtadminpb.GetWorkflowRequest{
+		ClusterId:  vars["cluster_id"],
+		Keyspace:   vars["keyspace"],
+		Name:       vars["name"],
+		ActiveOnly: activeOnly,
+	})
+
+	return NewJSONResponse(workflow, err)
+}
+
+// GetWorkflows implements the http wrapper for the VTAdminServer.GetWorkflows
+// method.
+//
+// Its route is /workflows, with query params:
+// - cluster: repeated, cluster IDs
+// - active_only
+// - keyspace: repeated
+// - ignore_keyspace: repeated
```

Comment:
Sometimes... I have _gross thoughts_ about somehow... adding protos for our HTTP request shapes, since I end up redefining the types on the front-end anyway. Probably our time would be better spent on gRPC-on-the-front-end... 😭 ..._unless_...............

Reply:
(my vote is also for grpc-on-the-front-end, some day ................)